sql/planner/optimizer.go

```diff
@@ -252,7 +252,7 @@ func RemoveUnnecessarySelectionNodesRule(t *Tree) (*Tree, error) {
 // UseIndexBasedOnSelectionNodeRule scans the tree for the first selection node whose condition is an
 // operator that satisfies the following criterias:
 // - implements the indexIteratorOperator interface
-// - one of its operands is a field selector that is indexed
+// - one of its operands is a path selector that is indexed
```

Comment:
We are talking about the `expr.FieldSelector` type, perhaps this would be better:

```suggestion
// - one of its operands is a expr.FieldSelector that is indexed
```